plugin/storage/es/spanstore/writer.go

```diff
@@ -107,15 +112,36 @@ func (s *SpanWriter) CreateTemplates(spanTemplate, serviceTemplate, indexPrefix
 	if indexPrefix != "" && !strings.HasSuffix(indexPrefix, "-") {
 		indexPrefix += "-"
 	}
-	_, err := s.client().CreateTemplate(indexPrefix + "jaeger-span").Body(spanTemplate).Do(context.Background())
-	if err != nil {
-		return err
-	}
-	_, err = s.client().CreateTemplate(indexPrefix + "jaeger-service").Body(serviceTemplate).Do(context.Background())
-	if err != nil {
-		return err
+	esVersion := s.client().GetVersion()
+	if esVersion > 7 {
+		spanTemplateResponse, err := s.v8client.Indices.PutIndexTemplate(indexPrefix+"jaeger-span", strings.NewReader(spanTemplate))
```

Comment:
nit: extract these two branches into helper functions createTemplatesV7 and createTemplatesV8

Reply:
Yes, missed that.